src/const_permutation.rs

```diff
@@ -0,0 +1,440 @@
+use core::fmt::Debug;
+
+use std::ops;
+
+#[cfg(feature = "random")]
+use rand::Rng;
+
+use crate::util::is_permutation;
+#[cfg(feature = "random")]
+use crate::Permutations;
+use crate::TryFromError;
+
+/// A permutation with a constant length fixed at compile time.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ConstPermutation<const N: usize>([usize; N]);
+impl<const N: usize> ConstPermutation<N> {
+    /// Returns the identity permutation.
+    pub fn identity() -> Self {
+        ConstPermutation((0..N).collect::<Vec<_>>().try_into().unwrap())
+    }
+    /// Returns the permutation which rotates r steps to the left.
+    pub fn rotation_left(r: usize) -> Self {
+        ConstPermutation(
+            (0..N)
+                .map(|i| (i + r) % N)
+                .collect::<Vec<_>>()
+                .try_into()
+                .unwrap(),
+        )
+    }
+    /// Returns the permutation which rotates r steps to the right.
+    pub fn rotation_right(r: usize) -> Self {
+        ConstPermutation::rotation_left(N - (r % N))
+    }
+    /// Returns the permutation which exchanges the elements at i and j.
+    pub fn transposition(i: usize, j: usize) -> Self {
+        assert!(i < N && j < N);
+        ConstPermutation(
+            (0..N)
+                .map(|k| {
+                    if k == i {
+                        j
+                    } else if k == j {
+                        i
+                    } else {
+                        k
+                    }
+                })
+                .collect::<Vec<_>>()
+                .try_into()
+                .unwrap(),
+        )
```

Comment:
I think we could reuse `identity` to make this

```suggestion
        let mut result = Self::identity();
        result.0.swap(i, j);
        
        result
```